dialoptions.go

```diff
@@ -474,6 +479,8 @@ func withBinaryLogger(bl binarylog.Logger) DialOption {
 //
 // Notice: This API is EXPERIMENTAL and may be changed or removed in a
 // later release.
```

Comment:
For this one, I'd say we should delete "experimental", but we probably want to reserve the right to remove it.  IIRC this makes its way through multiple layers of our library and might be more painful to maintain long-term.

Reply:
PTAL https://github.com/grpc/grpc-go/pull/7097/commits/cbdf3519fb67fb293b164642a2068b1502ca0451